app/src/common/shared/org/mozilla/vrbrowser/VRBrowserActivity.java

```diff
@@ -128,6 +129,21 @@ public void run() {
             }
         });
         initializeWorld();
+
+        int sessionId = SettingsStore.getInstance(this).getCurrentSessionId();
+        if (sessionId != SessionStore.NO_SESSION_ID) {
+            GeckoSession session = SessionStore.get().getSession(sessionId);
+            if (session != null) {
+                SessionStore.get().setCurrentSession(sessionId);
+                mPreviousSessionId = SettingsStore.getInstance(this).getPreviousSessionId();
+
+                if (session.getSettings().getBoolean(GeckoSessionSettings.USE_PRIVATE_MODE)) {
```

Comment:
The private browsing mode state setters are starting to get duplicated in different parts of the code. Could you use the SessionChangeListener.onCurrentSessionChange listener to handle the state and get rid of the setters in VRBrowserActivity?